authentik/events/models.py

```diff
@@ -331,6 +390,25 @@ def send(self, notification: "Notification") -> list[str]:
             return self.send_email(notification)
         raise ValueError(f"Invalid mode {self.mode} set")
 
+    def process_batch(self, event: Event) -> list[str]:
+        """Process an event for batching and send batch notification"""
+        batch = EventBatch.get_or_create_batch(event.action)
+        batch.add_event(event)
+
+        summary_message = batch.process_batch()
+        if summary_message:
+            batch.delete()
+            batch_notification = Notification(
+                severity=NotificationSeverity.NOTICE,
+                body=summary_message,
+                event=None,
```

Comment:
For the event data we can just use the most recent event, since iirc some webhook logic for the notifications uses the `event` field

Reply:
Done!